### PR TITLE
Add horizontal collapse

### PR DESCRIPTION
**Pull request by bestguy**
- [ ] Bug fix 
- [x] New feature 
- [ ] Chore 
- [ ] Breaking change 
- [ ] There is an open issue which this change addresses
- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [x] My commits follow the [Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [x] My change requires a change to [Typescript typings](./types/lib).
  - [x] I have updated the typings accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/Collapse.js [hunk 1]
```diff
@@ -117,10 +120,11 @@ class Collapse extends Component {
           let collapseClass = getTransitionClass(status);
           const classes = mapToCssModules(classNames(
             className,
+            horizontal && 'collapse-horizontal',
             collapseClass,
             navbar && 'navbar-collapse'
           ), cssModule);
-          const style = height === null ? null : { height };
+          const style = dimension === null ? null : { [horizontal ? 'width' : 'height']: dimension };
```
Comment (phwebi) [inline src/Collapse.js:127]: Can we add a test that the actual style is set correctly depending on whether the `horizontal` prop is true or false?

**Review by phwebi — COMMENTED** [review]
Nice!

File: src/Collapse.js (hunk 1 above)
Comment (bestguy) [reply src/Collapse.js:127]: Good idea, however this is not easy to test as the style attribute is removed after transition is over - the height and width are only present during the transition.

Comment (phwebi) [reply src/Collapse.js:127]: Gotcha - that's fine then!

**Review by phwebi — APPROVED** [review]

**Pull request merged**